core/src/gl/texture.cpp

```diff
@@ -189,6 +191,8 @@ void Texture::generate(GLuint _textureUnit) {
     GL_CHECK(glTexParameteri(m_target, GL_TEXTURE_WRAP_T, m_options.wrapping.wrapt));
 
     m_generation = RenderState::generation();
+
+    m_mainThreadJobQueue.makeCurrentThreadTarget();
```

Comment:
Ahh! I should have looked the implementation of that. Name looked familar to the evas call to set current gl thread :P. Thanks